### PR TITLE
feat(ui): add ability to unmount filesystems

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
## Done

- Added ability to unmount filesystems from the filesystems table.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the storage tab of a machine in Ready or Allocated state
- Change the storage layout to "LVM" which will create a partition and logical volume (a type of virtual disk)
- Check that you can unmount each filesystem, which puts them back into  "Available disks and partitions"

## Fixes

Fixes canonical-web-and-design/MAAS-squad#2275

### CODE REVIEW

File: ui/src/app/machines/views/MachineDetails/MachineStorage/FilesystemsTable/FilesystemsTable.test.tsx [hunk 1]
```diff
@@ -157,15 +157,70 @@ describe("FilesystemsTable", () => {
     expect(wrapper.find("AddSpecialFilesystem").exists()).toBe(true);
   });
 
-  it("can show an action to remove a filesystem", () => {
+  it("can remove a disk's filesystem", () => {
```
Comment (Caleb-Ellis) [inline ui/src/app/machines/views/MachineDetails/MachineStorage/FilesystemsTable/FilesystemsTable.test.tsx:160]: Driveby rewrite of tests for each type of filesystem removal, including action assertion

File: ui/src/app/store/machine/slice.ts [hunk 2]
```diff
@@ -533,8 +533,10 @@ const statusHandlers = generateStatusHandlers<
         }) => ({
           system_id: params.systemId,
           filesystem_id: params.filesystemId,
-          ...(params.blockDeviceId && { blockdevice_id: params.blockDeviceId }),
-          ...(params.partitionId && { partition_id: params.partitionId }),
+          ...("blockDeviceId" in params && {
+            blockdevice_id: params.blockDeviceId,
+          }),
+          ...("partitionId" in params && { partition_id: params.partitionId }),
```
Comment (Caleb-Ellis) [inline ui/src/app/store/machine/slice.ts:539]: Driveby fix for case where `params.partitionId` could be 0

**Review by squidsoup — APPROVED** [review]
Looks good, thanks caleb!
